pkg/sentry/kernel/fasync/fasync.go

```diff
@@ -96,14 +112,16 @@ func (a *FileAsync) Register(w waiter.Waitable) {
 // The file must be currently registered.
 func (a *FileAsync) Unregister(w waiter.Waitable) {
 	a.mu.Lock()
-	defer a.mu.Unlock()
+	a.regMu.Lock()
+	defer a.regMu.Unlock()
 
-	if a.e.Callback == nil {
+	if !a.registered {
```

Comment:
In Unregister function ,  a.registered need to be set false?

Reply:
Done.